### PR TITLE
Fix performance bug for NetworkGeometry::isLoadedWithTextures() and glTransform removal

### PR DESCRIPTION
**Pull request by samcake**
- introduced a cache boolean to remember once the textures of a model have been all loaded, this bool is invalidated whenever a texture gets reassigned.

- removed unused glTransform calls on the gpu::Batch

### CODE REVIEW

File: libraries/render-utils/src/GeometryCache.cpp [hunk 1]
```diff
@@ -1753,15 +1753,18 @@ bool NetworkGeometry::isLoadedWithTextures() const {
...
         return false;
     }
-    foreach (const NetworkMesh& mesh, _meshes) {
-        foreach (const NetworkMeshPart& part, mesh.parts) {
-            if ((part.diffuseTexture && !part.diffuseTexture->isLoaded()) ||
-                    (part.normalTexture && !part.normalTexture->isLoaded()) ||
-                    (part.specularTexture && !part.specularTexture->isLoaded()) ||
-                    (part.emissiveTexture && !part.emissiveTexture->isLoaded())) {
-                return false;
+    if (!_isLoadedWithTextures) {
```
Comment (ZappoMan) [inline libraries/render-utils/src/GeometryCache.cpp]: note: it is possible to programmatically change the textures.. I don't think you're handling that case

**Comment by hifi-gustavo** [comment]
Build console output for commit: a6f006a36d998cfd6450a66ec445e229c029483c available at: https://gist.github.com/f0fe43615680bc778826<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4267.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4267.dmg

File: libraries/render-utils/src/GeometryCache.cpp (hunk 1 above)
Comment (samcake) [inline libraries/render-utils/src/GeometryCache.cpp]: Isn't it covered by flipping the flag to false when calling the "setTextureWithNameToURL" ?
I thought that was the only place that could affect the status ?

Comment (ZappoMan) [inline libraries/render-utils/src/GeometryCache.cpp]: ah yes!

**Pull request merged**